docs/ref/reference.rst

```diff
@@ -85,225 +113,242 @@ nodes of the formation.
-     instance to monitor PostgreSQL services::
-
-      $ pg_autoctl create monitor --help
-      pg_autoctl create monitor: Initialize a pg_auto_failover monitor node
-      usage: pg_autoctl create monitor  [ --pgdata --pgport --pgctl --hostname ]
-
-        --pgctl       path to pg_ctl
-        --pgdata      path to data directory
-        --pgport      PostgreSQL's port number
-        --hostname    hostname by which postgres is reachable
-        --auth        authentication method for connections from data nodes
-        --skip-pg-hba skip editing pg_hba.conf rules
-        --run         create node then run pg_autoctl service
-
-     The ``--pgdata`` option is mandatory and default to the environment
-     variable ``PGDATA``. The ``--pgport`` default value is 5432, and the
-     ``--pgctl`` option defaults to the first ``pg_ctl`` entry found in your
-     `PATH`.
-
-     The ``--hostname`` option allows setting the hostname that the other
-     nodes of the cluster will use to access to the monitor. When not
-     provided, a default value is computed by running the following
-     algorithm:
-
-       1. Open a connection to the 8.8.8.8:53 public service and looks up the
-          TCP/IP client address that has been used to make that connection.
-
-       2. Do a reverse DNS lookup on this IP address to fetch a hostname for
-          our local machine.
+pg_autoctl create monitor
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This command initializes a PostgreSQL cluster and installs the
+`pgautofailover` extension so that it's possible to use the new instance to
+monitor PostgreSQL services::
+
+ $ pg_autoctl create monitor --help
+  pg_autoctl create monitor: Initialize a pg_auto_failover monitor node
+  usage: pg_autoctl create monitor  [ --pgdata --pgport --pgctl --hostname ]
+
+    --pgctl           path to pg_ctl
+    --pgdata          path to data directory
+    --pgport          PostgreSQL's port number
+    --hostname        hostname by which postgres is reachable
+    --auth            authentication method for connections from data nodes
+    --skip-pg-hba     skip editing pg_hba.conf rules
+    --run             create node then run pg_autoctl service
+    --ssl-self-signed setup network encryption using self signed certificates (does NOT protect against MITM)
+    --ssl-mode        use that sslmode in connection strings
+    --ssl-ca-file     set the Postgres ssl_ca_file to that file path
+    --ssl-crl-file    set the Postgres ssl_crl_file to that file path
+    --no-ssl          don't enable network encryption (NOT recommended, prefer --ssl-self-signed)
+    --server-key      set the Postgres ssl_key_file to that file path
+    --server-cert     set the Postgres ssl_cert_file to that file path
+
+The ``--pgdata`` option is mandatory and default to the environment variable
+``PGDATA``. The ``--pgport`` default value is 5432, and the ``--pgctl``
+option defaults to the first ``pg_ctl`` entry found in your `PATH`.
```

Comment:
On the monitor we don't really care, we have the same set of features in all supported Postgres versions so we take the first entry in the PATH:

``` c
	if (IS_EMPTY_STRING_BUFFER(options.pgSetup.pg_ctl))
	{
		set_first_pgctl(&(options.pgSetup));
	}
```

On the keeper side we do want the user to make an explicit choice when there are more than one `pg_ctl` found, because when picking the wrong one, then the application might not be working correctly, so we error out in that case.